setup_build.py

```diff
@@ -43,7 +43,9 @@ def localpath(*args):
    'libraries'      : ['hdf5', 'hdf5_hl'],
    'include_dirs'   : [localpath('lzf')],
    'library_dirs'   : [],
-   'define_macros'  : [('H5_USE_16_API', None)]
+   'define_macros'  : [('H5_USE_16_API', None),
+                       ('NPY_NO_DEPRECATED_API', 0),
+                      ]
```

Comment:
Theoretically this should be `NPY_1_7_API_VERSION` not `0` to take advantage of the "new" (as of 2010) NumPy interfaces, but that is the subject of a different PR. At least this way the annoying warning will not be printed. NumPy has no intention of removing the old API version interfaces, so this is harmless.